resolves #3 write content for documentation site home page (#4)
asciidoctor/asciidoctor-community-docs: home/modules/ROOT/pages/index.adoc

++++
endif::[]

Welcome to the Asciidoctor documentation site!
Here you can find the reference materials, guides, and examples to write and publish content in AsciiDoc using Asciidoctor.
This documentation will help you start your journey with AsciiDoc or dive deeper if you're already well on your way.

.xref:asciidoc::index.adoc[Learn about AsciiDoc]
[.panel]
--
AsciiDoc is a plain text language for writing content such as documentation, articles, and blog posts.
If you're just starting out with AsciiDoc and want to discover what it's all about, head over to the xref:asciidoc::index.adoc[documentation for the AsciiDoc Language].
You'll learn everything from how to make your first AsciiDoc document to how to use advanced language features such as list continuations and tailoring substitutions.

If you need a condensed refresher, or want a glimpse of what AsciiDoc has to offer, check out the xref:asciidoc::syntax-quick-reference.adoc[syntax quick reference].
You can also find a side-by-side xref:asciidoc::asciidoc-vs-markdown.adoc[comparison with Markdown] if you're looking to upgrade.

The language documentation is primarily about the source you type to compose a document.

> One you're done writing and want to publish your AsciiDoc document, you'll need an AsciiDoc processor.

Author: ```suggestion
One you're done writing and want to publish your AsciiDoc document, you'll need an AsciiDoc processor.
That's where Asciidoctor fits in.
```

--

//.xref:asciidoctor::index.adoc[Publish AsciiDoc content]
//.xref:asciidoctor::index.adoc[Process AsciiDoc content]
//.xref:asciidoctor::index.adoc[AsciiDoc processing]
.xref:asciidoctor::index.adoc[Process your AsciiDoc content]
[.panel]
--

> That's where Asciidoctor comes in.

Author: ```suggestion
```

xref:asciidoctor.js::index.adoc[Asciidoctor.js (JavaScript)]::
Transpiles Asciidoctor into JavaScript so you can run it in the browser or a Node.js application.
It provides a CLI and a porcelain API for methods in the Asciidoctor document model and enables you to write Asciidoctor extensions in JavaScript.

[discrete#explore]
=== Explore Asciidoctor

[.grid]
Build automation::
Convert your content from AsciiDoc automatically when you run your build.
Encapsulate assets, configuration profiles, and extensions for your publishing process in your build scripts.
Even convert content in your source code or test suite.

* xref:maven-tools::index.adoc[Maven tools]
* *Gradle plugin*
* *Asciidoclet (for Javadoc)*

Slide presentations::
Author HTML slide presentations in AsciiDoc without having to wrestle with the clutter of HTML.
Reserve the option of single sourcing your content for publishing to other formats.

* xref:reveal.js-converter::index.adoc[reveal.js converter]

Site generation::
Want to build a documentation site like this one?
AsciiDoc can be used as the source format for the content of a site.

> An AsciiDoc processor, such as Asciidoctor.js, and be use to integrate that content into a full site build.

Author: ```suggestion
A static site generator can delegate to an AsciiDoc processor such as Asciidoctor.js to convert AsciiDoc files in the site source to embedded HTML, then apply a template to that output to make pages for the site.
```

Browse the links below or the navigation on the left to locate the information you need.

.xref:asciidoc::index.adoc[Learn about AsciiDoc]
[.panel]
--
AsciiDoc is a plain text language for writing content such as documentation, articles, and blog posts.
If you're just starting out with AsciiDoc, or you want to discover what it's all about, head over to the xref:asciidoc::index.adoc[documentation for the AsciiDoc Language].
You'll learn everything from how to make your first AsciiDoc document to how to use advanced language features such as list continuations and tailoring substitutions.

If you just need a quick refresher, or want to see a glimpse of what AsciiDoc has to offer, check out the xref:asciidoc::syntax-quick-reference.adoc[syntax quick reference].
You can also find a side-by-side xref:asciidoc::asciidoc-vs-markdown.adoc[comparison with Markdown] if you're looking to upgrade.

The language documentation is primarily about the source you type to compose a document.
One you're done writing and want to publish the AsciiDoc for others to read, you'll need an AsciiDoc processor.
--

> .xref:asciidoctor::index.adoc[Publish with AsciiDoc]

Reviewer: As a non-native speaker, it sounds like AsciiDoc is a tool. Maybe "Publish (your) AsciiDoc documents"?

Author: I had a similar thought while I was sleeping on it. How about "Publish AsciiDoc content"?

Reviewer: I like your proposal, "Publish AsciiDoc content" is clear and concise 👌🏻

That's where Asciidoctor comes in.

xref:asciidoctor::index.adoc[Asciidoctor] is an AsciiDoc processor.
That means Asciidoctor reads the AsciiDoc source you entered, parses it into document model, and converts it to another format such as HTML.
Asciidoctor comes with both a xref:asciidoctor:cli:index.adoc[CLI] and xref:asciidoctor:api:index.adoc[API] that you can use to either invoke one of the built-in converters (HTML, DocBook, man page) or an extended converter.
Asciidoctor also enriches the HTML by applying a xref:asciidoctor:html-backend:default-stylesheet.adoc[default stylesheet], adding stylistic icons, and xref:asciidoctor:syntax-highlighting:index.adoc[syntax highlighting] source blocks.
--

[discrete#chose]
=== Choose Your Runtime

There are three variants of Asciidoctor (i.e., the core AsciiDoc processor), all of which share the same code base.
With few exceptions, these variants all provide the full Asciidoctor experience.
From these three shared implementations extend a whole host of extensions, build tool integrations, and tooling plugins.
Choose the one that best suits the language platform(s) you're using.

> [.grid]

Reviewer: Maybe we should wrap on a small screen it gets very narrow

![wrap](https://user-images.githubusercontent.com/333276/102092020-ddbd2180-3e1f-11eb-8bf4-84dd1de66ac4.png)

Author: I haven't done the mobile styles yet. They will drop back to a normal dlist layout.

xref:asciidoctor::index.adoc[Asciidoctor (Ruby)]::
The original code base, which enables you to run Asciidoctor using any Ruby implementation, including C Ruby, JRuby, and TruffleRuby.
Both JRuby and TruffleRuby allow you to use Asciidoctor on a JVM.

AsciidoctorJ (Java)::
A Java library that encapsulates the use of JRuby to load and run Asciidoctor on the JVM.
Provides a CLI as well as native Java API wrappers for all the Asciidoctor APIs and enables you to write Asciidoctor extensions in Java.

> xref:asciidoctor.js::index.adoc[Asciidoctor.js] (JS)::

Reviewer: I think we should use the complete name:

```suggestion
xref:asciidoctor.js::index.adoc[Asciidoctor.js] (JavaScript)::
```

Author: Oops, yes, the link should go all the way through. I used JS since that's what we use in the topbar nav to save space. I suppose since we have the space here we should spell it out.

Transpiles Asciidoctor into JavaScript so you can run it in the browser or in a Node.js application.
Provides a CLI as well as a porcelain API for methods in the Asciidoctor document model and enables you to write Asciidoctor extensions in JavaScript.

[discrete#explore]
=== Explore Asciidoctor

> [.grid]

Reviewer: ditto

![wrap1](https://user-images.githubusercontent.com/333276/102092764-b024a800-3e20-11eb-9b3e-174844caea1f.png)

Build automation::
Convert your content from AsciiDoc automatically when you run your build.
Encapsulate assets, configuration profiles, and extensions for your publishing process in your build scripts.
Even convert content in your source code or test suite.

* xref:maven-tools::index.adoc[Maven tools]
* *Gradle plugin*
* *Asciidoclet (for Javadoc)*

Slide presentations::
Author HTML slide presentations in AsciiDoc without having to wrestle with the clutter of HTML.
Reserve the option of single sourcing your content for publishing to other formats.

* xref:reveal.js-converter::index.adoc[reveal.js converter]

Site generation::
Want to build a documentation site like this one?
AsciiDoc can be used as the source format for the content of a site.
An AsciiDoc processor, such as Asciidoctor.js, and be use to integrate that content into a full site build.

* https://antora.org[Antora^]

Essential reference::

> Don't miss these essential reference material to help you write and publish content with AsciiDoc.

Reviewer: "these essential "... we should add a "s" somewhere no? (multiple essential references?)

= Asciidoctor Documentation
:navtitle: Home
:page-role: home

ifdef::env-site[]
++++
<style>
.doc .content > div:first-child {
  margin-top: 0;
}

.doc .panel {
  margin-top: 1rem;
  background-color: #f5f5f5;
  border-radius: 0.5em;
  padding: 0.75rem 1.25rem;
  font-size: 0.8888rem;
}

.doc .panel .title {
  font-size: 1rem;
  font-style: normal;
  letter-spacing: 0;
  padding-bottom: 0.25rem;
}

h3.discrete {
  margin-top: 2rem;
}

.doc .grid {
  margin-top: 1rem;
  font-size: 0.8333rem;
}

.doc .grid dl {
  display: grid;
  grid-auto-columns: 1fr;
  grid-auto-flow: row;
  grid-gap: 0.25rem 1rem;
}

.doc .grid dt {
  grid-row: 1;
  font-style: normal;
  padding-right: 0;
}

.doc .grid dd {
  margin: 0;
}

.doc .grid .ulist {
  margin-top: 0;
  line-height: 1.3;
}

.doc .grid ul {
  padding-left: 0;
  list-style: none;
}

.doc .grid ul li {
  margin-top: 0.15em;
  margin-bottom: 0;
  hyphens: none;
}

.doc .grid ul a {
  font-weight: bold;
}
</style>
++++
endif::[]

Welcome to the Asciidoctor documentation site!

> Here you can find all the reference material, guides, and examples you need to write and publish content in AsciiDoc using Asciidoctor.

Reviewer 2: ```suggestion
Here you can find the reference materials, guides, and examples to write and publish content in AsciiDoc using Asciidoctor.
```

> This documentation will help you get started on your journey with AsciiDoc or dive deeper if you're already well on your way.

Reviewer 2: ```suggestion
This documentation will help you start your journey with AsciiDoc or dive deeper if you're already well on your way.
```

> Browse the links below or the navigation on the left to locate the information you need.

Reviewer 2: ```suggestion

```

.xref:asciidoc::index.adoc[Learn about AsciiDoc]
[.panel]
--
AsciiDoc is a plain text language for writing content such as documentation, articles, and blog posts.

> If you're just starting out with AsciiDoc, or you want to discover what it's all about, head over to the xref:asciidoc::index.adoc[documentation for the AsciiDoc Language].

Reviewer 2: ```suggestion
If you're just starting out with AsciiDoc and want to discover what it's all about, head over to the xref:asciidoc::index.adoc[documentation for the AsciiDoc Language].
```

You'll learn everything from how to make your first AsciiDoc document to how to use advanced language features such as list continuations and tailoring substitutions.

> If you just need a quick refresher, or want to see a glimpse of what AsciiDoc has to offer, check out the xref:asciidoc::syntax-quick-reference.adoc[syntax quick reference].

Reviewer 2: ```suggestion
If you need a condensed refresher, or want a glimpse of what AsciiDoc has to offer, check out the xref:asciidoc::syntax-quick-reference.adoc[syntax quick reference].
```

You can also find a side-by-side xref:asciidoc::asciidoc-vs-markdown.adoc[comparison with Markdown] if you're looking to upgrade.

The language documentation is primarily about the source you type to compose a document.

> One you're done writing and want to publish the AsciiDoc for others to read, you'll need an AsciiDoc processor.

Reviewer 2: ```suggestion
One you're done writing and want to publish your AsciiDoc document, you'll need an AsciiDoc processor.
```

--

//.xref:asciidoctor::index.adoc[Publish AsciiDoc content]
//.xref:asciidoctor::index.adoc[Process AsciiDoc content]
//.xref:asciidoctor::index.adoc[AsciiDoc processing]
.xref:asciidoctor::index.adoc[Process your AsciiDoc content]
[.panel]
--

> AsciiDoc is a shorthand format that's focused on maintaining your writing flow.

Reviewer 2: ```suggestion

```

> AsciiDoc is a shorthand format that's focused on maintaining your writing flow.

Reviewer 2: I'd delete lines 102 - 104 (and move 105). I don't think they're helpful and a bit repetitive.

> While the format is great for collaborative writing and editing, it's not intended for publishing.

Reviewer 2: ```suggestion

```

> When it comes time to publish your AsciiDoc content for others to read, you'll need to transform it to another format.

Reviewer 2: ```suggestion

```

> That's where Asciidoctor comes in.

Reviewer 2: I'd move this line to right below line 93 in the previous panel.

> xref:asciidoctor::index.adoc[Asciidoctor] is an AsciiDoc processor.

Reviewer 2: ```suggestion
xref:asciidoctor::index.adoc[Asciidoctor] is the core AsciiDoc processor.
```
I'm making this suggestion because I'm not super excited about putting the "(i.e., the core AsciiDoc processor)" in the runtime panel and would rather add this nuance here.

> That means Asciidoctor reads the AsciiDoc source you entered, parses it into document model, and converts it to another format such as HTML.

Reviewer 2: ```suggestion
It reads the AsciiDoc source you wrote, parses it into a document model, and converts it to another format such as HTML.
```

> Asciidoctor comes with both a xref:asciidoctor:cli:index.adoc[CLI] and xref:asciidoctor:api:index.adoc[API] that you can use to either invoke one of the built-in converters (HTML, DocBook, man page) or an extended converter.

Reviewer 2: ```suggestion
Asciidoctor has a xref:asciidoctor:cli:index.adoc[CLI] and xref:asciidoctor:api:index.adoc[API] that you can use to invoke a built-in converter (HTML, DocBook, man page) or an extended converter.
```

> Asciidoctor also enriches the HTML by applying a xref:asciidoctor:html-backend:default-stylesheet.adoc[default stylesheet], adding stylistic icons, and xref:asciidoctor:syntax-highlighting:index.adoc[syntax highlighting] source blocks.

Reviewer 2: ```suggestion
Asciidoctor enriches the HTML it produces from an AsciiDoc document by applying a xref:asciidoctor:html-backend:default-stylesheet.adoc[default stylesheet], adding stylistic icons, and xref:asciidoctor:syntax-highlighting:index.adoc[syntax highlighting] source blocks.
```

--

[discrete#chose]
=== Choose Your Runtime

> There are three variants of Asciidoctor (i.e., the core AsciiDoc processor), all of which share the same code base.

Reviewer 2: ```suggestion
There are three variants of Asciidoctor that share the same code base.
```

With few exceptions, these variants all provide the full Asciidoctor experience.

> From these three shared implementations extend a whole host of extensions, build tool integrations, and tooling plugins.

Reviewer 2: ```suggestion
From these three implementations extend a host of extensions, build tool integrations, and tooling plugins.
```

> Choose the one that best suits the language platform(s) you're using.

Reviewer 2: ```suggestion
Choose the one that best suits the language platform you're using.
```

[.grid]
xref:asciidoctor::index.adoc[Asciidoctor (Ruby)]::

> The original code base, which enables you to run Asciidoctor using any Ruby implementation, including C Ruby, JRuby, and TruffleRuby.

Reviewer 2: ```suggestion
The original code base enables you to run Asciidoctor using any Ruby implementation, including C Ruby, JRuby, and TruffleRuby.
```

> Both JRuby and TruffleRuby allow you to use Asciidoctor on a JVM.

Reviewer 2: ```suggestion
JRuby and TruffleRuby allow you to use Asciidoctor on a JVM.
```

AsciidoctorJ (Java)::
A Java library that encapsulates the use of JRuby to load and run Asciidoctor on the JVM.

> Provides a CLI as well as native Java API wrappers for all the Asciidoctor APIs and enables you to write Asciidoctor extensions in Java.

Reviewer 2: ```suggestion
It provides a CLI and native Java API wrappers for all of the Asciidoctor APIs and enables you to write Asciidoctor extensions in Java.
```

xref:asciidoctor.js::index.adoc[Asciidoctor.js (JavaScript)]::

> Transpiles Asciidoctor into JavaScript so you can run it in the browser or in a Node.js application.

Reviewer 2: ```suggestion
Transpiles Asciidoctor into JavaScript so you can run it in the browser or a Node.js application.
```

> Provides a CLI as well as a porcelain API for methods in the Asciidoctor document model and enables you to write Asciidoctor extensions in JavaScript.

Reviewer 2: ```suggestion
It provides a CLI and a porcelain API for methods in the Asciidoctor document model and enables you to write Asciidoctor extensions in JavaScript.
```

[discrete#explore]
=== Explore Asciidoctor

[.grid]
Build automation::
Convert your content from AsciiDoc automatically when you run your build.
Encapsulate assets, configuration profiles, and extensions for your publishing process in your build scripts.
Even convert content in your source code or test suite.

* xref:maven-tools::index.adoc[Maven tools]
* *Gradle plugin*
* *Asciidoclet (for Javadoc)*

Slide presentations::
Author HTML slide presentations in AsciiDoc without having to wrestle with the clutter of HTML.
Reserve the option of single sourcing your content for publishing to other formats.

* xref:reveal.js-converter::index.adoc[reveal.js converter]

Site generation::
Want to build a documentation site like this one?
AsciiDoc can be used as the source format for the content of a site.

> An AsciiDoc processor, such as Asciidoctor.js, and be use to integrate that content into a full site build.

Reviewer 2: This sentence doesn't quite make sense...maybe it's missing a verb or noun (basically some words seem to be missing or wrong)?

Author: Yep, typos abound!

* https://antora.org[Antora^]

Essential reference::

> Don't miss this essential reference material to help you write and publish content with AsciiDoc.

Reviewer 2: ```suggestion
Don't miss these essential reference materials for writing and publishing content with AsciiDoc and Asciidoctor.
```

* xref:asciidoc::syntax-quick-reference.adoc[AsciiDoc syntax quick reference]

> * xref:asciidoc:attributes:document-attributes-reference.adoc[Built-in document attributes]

Reviewer 2: ```suggestion
* xref:asciidoc:attributes:document-attributes-reference.adoc[Document attributes]
```
